pkg/operator/clustermemberremovalcontroller/clustermemberremovalcontroller.go

```diff
@@ -132,6 +132,10 @@ func (c *clusterMemberRemovalController) attemptToScaleDown(ctx context.Context,
 		return fmt.Errorf("could not list etcd members: %w", err)
 	}
 
+	// removing etcd-boostrap from live voting members
+	// see https://issues.redhat.com/browse/OCPBUGS-10351
+	liveVotingMembers = removeBootstrapMember(liveVotingMembers)
```

Comment:
Do we want similar filtering [down here][1] to remove a possible bootstrap member from `healthyLiveVotingMembers`, [the quorum check][2], and [the all-healthy check][3]?

[1]: https://github.com/openshift/cluster-etcd-operator/blob/0584b0d1c8868535baf889d8c199f605aef4a3ae/pkg/operator/clustermemberremovalcontroller/clustermemberremovalcontroller.go#L158-L162
[2]: https://github.com/openshift/cluster-etcd-operator/blob/0584b0d1c8868535baf889d8c199f605aef4a3ae/pkg/operator/clustermemberremovalcontroller/clustermemberremovalcontroller.go#L171
[3]: https://github.com/openshift/cluster-etcd-operator/blob/0584b0d1c8868535baf889d8c199f605aef4a3ae/pkg/operator/clustermemberremovalcontroller/clustermemberremovalcontroller.go#L216

Reply:
+1, thats why i wanted to skip scaling down as long as `etcd-boostrap` exist, imho it is cleaner fix :) 

wdyt @hasbro17 @tjungblu

Reply:
I agree  on skipping scale-down until we have a full set of non-bootstrap voting members.  But the presence of such doesn't mean the non-bootstrap voting members are all healthy yet.  I was suggesting something like:

```diff
--- a/pkg/operator/clustermemberremovalcontroller/clustermemberremovalcontroller.go
+++ b/pkg/operator/clustermemberremovalcontroller/clustermemberremovalcontroller.go
@@ -165,6 +165,12 @@ func (c *clusterMemberRemovalController) attemptToScaleDown(ctx context.Context,
                return fmt.Errorf("could not list etcd healthy members: %w", err)
        }
 
+       // We pruned bootstrap from liveVotingMembers above.  Prune
+       // healthyVotingMembers the same way, so we can compare them, and
+       // don't count a healthy bootstrap member when comparing with
+       // minTolerableQuorum or desiredControlPlaneReplicasCount.
+       healthyVotingMembers = removeBootstrapMember(healthyVotingMembers)
+
        // scaling down invariant
        minTolerableQuorum, err := etcdcli.MinimumTolerableQuorum(desiredControlPlaneReplicasCount)
        if err != nil {
```

But I also don't have a solid grasp on this logic, so it's certainly possible I'm missing something, and your pivot is fine as it stands.

Reply:
done